apps/server/src/modules/oauth/controller/oauth-sso.controller.ts

```diff
@@ -57,6 +58,25 @@ export class OauthSSOController {
 		res.redirect(errorRedirect.toString());
 	}
 
+	private migrationErrorHandler(error: unknown, session: ISession, res: Response) {
+		this.logger.error(error);
+		const migrationError: OAuthMigrationError =
+			error instanceof OAuthMigrationError ? error : new OAuthMigrationError();
+
+		session.destroy((err) => {
+			this.logger.log(err);
+		});
+
+		const errorRedirect: URL = new URL('/migration/error', this.clientUrl);
+		errorRedirect.searchParams.append('errorcode', migrationError.errorcode);
+		errorRedirect.searchParams.append('sourceSystem', <string>migrationError.sourceSystemId);
+		errorRedirect.searchParams.append('targetSystem', <string>migrationError.targetSystemId);
+		errorRedirect.searchParams.append('sourceSchoolNumber', <string>migrationError.officialSchoolNumberFromSource);
+		errorRedirect.searchParams.append('targetSchoolNumber', <string>migrationError.officialSchoolNumberFromTarget);
+
+		res.redirect(errorRedirect.toString());
+	}
```

Comment:
You can simplify this:

private migrationErrorHandler(error: unknown, session: ISession, res: Response) {
    this.logger.error(error);

    const migrationError = error instanceof OAuthMigrationError
        ? error
        : new OAuthMigrationError();

    session.destroy((err) => this.logger.log(err));

    const queryParams = new URLSearchParams({
        errorcode: migrationError.errorcode,
        sourceSystem: migrationError.sourceSystemId?.toString(),
        targetSystem: migrationError.targetSystemId?.toString(),
        sourceSchoolNumber: migrationError.officialSchoolNumberFromSource?.toString(),
        targetSchoolNumber: migrationError.officialSchoolNumberFromTarget?.toString(),
    });

    const errorRedirect = new URL('/migration/error', this.clientUrl);
    errorRedirect.search = queryParams.toString();

    res.redirect(errorRedirect.toString());
}

Reply:
this doesn't work, if we want to "simplify" it will be just more code and it will be illegible..